packages/langium/src/parser/token-builder.ts

```diff
@@ -8,11 +8,11 @@ import { Lexer, TokenPattern, TokenType } from 'chevrotain';
 import { terminalRegex } from '..';
 import { Grammar, isKeyword, isTerminalRule, Keyword, TerminalRule } from '../grammar/generated/ast';
 import { streamAllContents } from '../utils/ast-util';
-import { partialMatches } from '../utils/regex-util';
+import { getCaseInsensitivePattern, partialMatches } from '../utils/regex-util';
 import { stream } from '../utils/stream';
 
 export interface TokenBuilder {
-    buildTokens(grammar: Grammar): TokenType[];
+    buildTokens(grammar: Grammar, caseInsensitive?: boolean): TokenType[];
```

Comment:
I'd propose to wrap this in an object:
```suggestion
    buildTokens(grammar: Grammar, options?: { caseInsensitive?: boolean }): TokenType[];
```
This is better extensible and avoids a Boolean function parameter, which I regard as bad API in general.